magento/pwa-studio
# [PWA-370] Handling isFastNetwork.

## Description

`isFastNetwork` is a utility used in the service worker to understand if a network is fast enough to perform actions like prefetch. It uses `network.connection.effectiveType` internally. Unfortunately, this API is not adopted by all browsers at this point. Hence the function was defaulting to `false` in browsers like firefox which do not support the network API yet.

I tried to use utilities to calculate network bandwidth both internal (performance API / writing own speed calculator) and external (network-speed) and decided to stick to default to `true` for now. Firefox and safari should start supporting the network API soon. 

## Related Issue
Closes #2161 

### Verification Stakeholders
@larsroettig 

### Verification Steps
1. Go to any product page. (Make sure the SW is active and working)
2. Images should be prefetched in firefox.

## Screenshots / Screen Captures (if appropriate)


## Checklist
None
